datatypes/src/collections/data_types.rs

```diff
@@ -9,3 +12,37 @@ pub enum VectorDataType {
     MultiLineString,
     MultiPolygon,
 }
+
+#[derive(Debug)]
+pub enum TypedFeatureCollection {
+    Data(DataCollection),
+    MultiPoint(MultiPointCollection),
+    MultiLineString(MultiLineStringCollection),
+    MultiPolygon(MultiPolygonCollection),
+}
+
+impl TypedFeatureCollection {
```

Comment:
We cloud also add a trait for this and implement it for the types similar to RasterDAtaType

Reply:
please elaborate how this would help here

Reply:
just an idea.